gh-114070: correct the specification of ``digit`` in the float() docs (#114080)
python/cpython: Doc/library/functions.rst

      sign: "+" | "-"
      infinity: "Infinity" | "inf"
      nan: "nan"

>       digit: <a Unicode decimal digit, i.e. all characters the Unicode general category Nd>

Author: Hmm, maybe should be shortened a bit?
```suggestion
      digit: <a decimal digit, i.e. characters in the Unicode general category Nd>
```

Reviewer: I merged the grammar fix you suggested, but left the Unicode in. I think it's an important part of `digit` that's also easy to forget (e.g. https://github.com/python/cpython/issues/114070#issuecomment-1891424747) so doesn't hurt to emphasise. I'm not too worried about brevity here.

>       digit: <a Unicode decimal digit, i.e. all characters the Unicode general category Nd>

Reviewer: ```suggestion
      digit: <a Unicode decimal digit, i.e. characters in Unicode general category Nd>
```
